prog/types.go

```diff
@@ -197,7 +197,7 @@ func (t *IntTypeCommon) Format() BinaryFormat {
 // Returns the size in bits for integers in binary format or 64 for string-formatted integers. The return
 // value is used in computing limits and truncating other values.
 func (t *IntTypeCommon) TypeBitSize() uint64 {
-	if t.ArgFormat != FormatNative && t.ArgFormat != FormatBigEndian {
+	if t.ArgFormat != FormatNative && t.ArgFormat != FormatBigEndian || t.TypeSize > 8 {
```

Comment:
When is this new check needed?

Reply:
When we have a ConstType with **IsPad** = **true**: `&prog.ConstType{IntTypeCommon:prog.IntTypeCommon{TypeCommon:prog.TypeCommon{TypeName:"pad", FldName:"", TypeSize:0x70, ArgDir:0, IsOptional:false, IsVarlen:false}, ArgFormat:0, BitfieldOff:0x0, BitfieldLen:0x0, BitfieldMdl:false}, Val:0x0, IsPad:true}`.
I am thinking to write a particular implementation for `ConstType`, but I will have some duplicated lines with the one for `IntTypeCommon`. Do you have any other suggestions?

Reply:
Use IsPad function to detect padding.
However, why/where are we trying to query TypeBitSize for padding? We should not generate/mutate it. I think we should panic if somebody asks for TypeBitSize padding. Add the panic and see how is calling it and why. Perhaps there is a bug in the caller.